main/src/cgeo/geocaching/models/Geocache.java

```diff
@@ -1558,7 +1558,7 @@ public boolean addWaypointsFromNote() {
      */
     public boolean addWaypointsFromText(@Nullable final String text, final boolean updateDb, @NonNull final String namePrefix, final boolean forceExtraction) {
         boolean changed = false;
-        if (forceExtraction || !preventWaypointsFromNote) {
+        if (forceExtraction || (!Settings.isGlobalWpExtractionDisabled() && !preventWaypointsFromNote)) {
```

Comment:
Not sure if I understand this condition correctly. To my understanding we wanted to implement an opt-out system. Wouldn't we need to check this in the `Geocache` class then instead of here? Distinguish on cache-level between a manually set value (true/false), which would override the global setting, and a "not yet set" value, which would activate the global setting for this cache?

Reply:
> To my understanding we wanted to implement an opt-out system.

Cannot judge the code, but indeed an opt-out system was the idea:
- If the setting is not active (which should be default) waypoints are extracted and only surpressed if the user activates the checkbox under the specific personal note.
- It the setting is active (set by user, no default) the checkbox should be activated by default for all caches and can be individually deactivated by the user for each cache.

Reply:
> > To my understanding we wanted to implement an opt-out system.
> 
> Cannot judge the code, but indeed an opt-out system was the idea:
> 
> * If the setting is not active (which should be default) waypoints are extracted and only surpressed if the user activates the checkbox under the specific personal note.

That is indeed the implemented behaviour

> * It the setting is active (set by user, no default) the checkbox should be activated by default for all caches and can be individually deactivated by the user for each cache.

Ahh ok, that is not what the change does at the moment, however, I agree that is a much better. I understand that this type of change does not go into the place I put it. I have not found out yet, where in the code I could set the checkbox you mentioned but I will investigate that tomorrow.

Once I have it done, would I do another push into the same branch?

Reply:
> Ahh ok, that is not what the change does at the moment, however, I agree that is a much better. I understand that this type of change does not go into the place I put it. I have not found out yet, where in the code I could set the checkbox you mentioned but I will investigate that tomorrow.

Database change would go into `storage/DataStore`, applying the default value to `models/Geocache` (IIRC).

> Once I have it done, would I do another push into the same branch?

Yes, for the time being. (At then end, when everything is fine, we would squash the individual commits into one commit containing all the changes necessary for this PR.)

Thanks for working on this issue! Very welcomed!

Reply:
> > * It the setting is active (set by user, no default) the checkbox should be activated by default for all caches and can be individually deactivated by the user for each cache.
> 
> Ahh ok, that is not what the change does at the moment, however, I agree that is a much better. I understand that this type of change does not go into the place I put it. I have not found out yet, where in the code I could set the checkbox you mentioned but I will investigate that tomorrow.

Sorry for late comment..
Imho, the easiest way to achieve the desired behaviour is to set the default setting of `Geocache.preventWaypointsFromNote` to the new setting. Meaning: in `Geocache.java`, change line 176:
`    private boolean preventWaypointsFromNote = false;`

to

`    private boolean preventWaypointsFromNote = Settings.isGlobalWpExtractionDisabled();`

That way the checkbox handling is automatically "done", and also the value would be correctly overridden from Database in case the cache is already stored there with a different setting.

Reply:
> > > * It the setting is active (set by user, no default) the checkbox should be activated by default for all caches and can be individually deactivated by the user for each cache.
> > 
> > 
> > Ahh ok, that is not what the change does at the moment, however, I agree that is a much better. I understand that this type of change does not go into the place I put it. I have not found out yet, where in the code I could set the checkbox you mentioned but I will investigate that tomorrow.
> 
> Sorry for late comment..
> Imho, the easiest way to achieve the desired behaviour is to set the default setting of `Geocache.preventWaypointsFromNote` to the new setting. Meaning: in `Geocache.java`, change line 176:
> ` private boolean preventWaypointsFromNote = false;`
> 
> to
> 
> ` private boolean preventWaypointsFromNote = Settings.isGlobalWpExtractionDisabled();`
> 
> That way the checkbox handling is automatically "done", and also the value would be correctly overridden from Database in case the cache is already stored there with a different setting.

Hi Eddie, you wouldn't believe it, that is **exactly** the change I just made ;) I have started to test it and it works as expected and as discussed in this thread yesterday.

I am going to push it shortly. Please have a look at the code and prefereably I'd also want some of you to give it a real try and walk through some usage scenarios on the device.

Reply:
I have just committed it into yesterday's pull request. I hope this was the right way doing it from a process point of view.